rest/oidc_api.go

```diff
@@ -82,7 +107,13 @@ func (h *handler) handleOIDCCommon() (redirectURLString string, err error) {
 	}
 
 	var redirectURL *url.URL
-	state := ""
+	var state string
+
+	// Set state parameter to prevent cross-site request forgery (CSRF) when DisableCallbackState is not enabled.
+	if provider.DisableCallbackState != nil && !*provider.DisableCallbackState {
+		state = base.GenerateRandomSecret()
+		setStateCookie(h.response, state)
```

Comment:
Not sure it's nessesary to declare the variable outside of the scope here.

```suggestion
	// Set state parameter to prevent cross-site request forgery (CSRF) when DisableCallbackState is not enabled.
	if provider.DisableCallbackState != nil && !*provider.DisableCallbackState {
		setStateCookie(h.response, base.GenerateRandomSecret())
```

Reply:
Ok I see now. I'd missed the redirectURL building directly below this. Makes sense.